test/foss/test_foss.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2023 Ericsson AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import shlex
+import subprocess
+import sys
+import unittest
+import os
+
+ROOT_DIR = "./"
+NOT_PROJECT_FOLDERS = ["templates", "__pycache__", ".pytest_cache"]
+
+
+def get_dynamic_test_dirs():
+    dirs = []
+    for entry in os.listdir(ROOT_DIR):
+        full_path = os.path.join(ROOT_DIR, entry)
+        if os.path.isdir(full_path) and entry not in NOT_PROJECT_FOLDERS:
+            dirs.append(entry)
+    return dirs
+
+
+PROJECT_DIRS = get_dynamic_test_dirs()
+
+
+# This will contain the generated tests.
+# I have not used the common lib from unit test, because it would
+# greatly increase the difficulty of the implementation, and this way the
+# two test aren't bound to each other
```

Comment:
Don't use first person. Can't we just put the common library under `test` instead of `test/unit`? My understanding is that CodeChecker does that.